### DISCUSSION TITLE
AWS Status feed

### DISCUSSION THREAD

**Discussion by yesoreyeram** [grafana/grafana-infinity-datasource#244] [Specific APIs]
## AWS Status Feeds in Grafana

In this example, we are going to see how we use use AWS status feeds as Grafana annotations. This will be useful when you are dealing with any AWS outages. For this, we are going to use the [AWS RSS feed](https://status.aws.amazon.com/rss/all.rss). You can find more feeds at [AWS Status page](https://status.aws.amazon.com/)

![image](https://user-images.githubusercontent.com/153843/151575227-20088546-4368-4066-a91b-64058982544b.png#center)

## Connection setup

AWS status are open and no authentication required. So, you can simple create a datasource using infinity without any additional configuration.

## Annotation setup

Once you create a annotation, you will perform the following steps

* Create a annotation and select your Infinity datasource
* Select "XML" as query type, "URL" as source and Format "Data Frame"
* Provide `https://status.aws.amazon.com/rss/all.rss` as the URL
* You need to specify `rss.channel[0].item` as the URL. ( You can find this path from the original rss feed )
* Create `title`, `description` as columns and provide string type
* Create `pubDate` column and mark this as 'DateTime'
* Finally select `guid[0]._` as string. This is your link. So you can alias this to link

![image](https://user-images.githubusercontent.com/153843/151575928-4fc9f188-7f9a-43c5-a92a-6069fe434e6a.png)

## Table view of status items

You can follow the same query procedure in your table panel to get the results as Table

![image](https://user-images.githubusercontent.com/153843/151576874-6f4d73d2-9331-4473-a7aa-a3eae0bec880.png)

## Alternate query method

If you are familiar with UQL query, you can achieve this with a simple query. Instead of select "XML" as your query type, you will choose "UQL" in this method and write the following UQL query.

![image](https://user-images.githubusercontent.com/153843/151577609-d2e5a7c3-aaf8-412b-83b8-965ca676eef4.png)

```sql
parse-xml
| scope "rss.channel.item"
| extend "published date"=todatetime("pubDate")
| project "title", "published date", "description", "link"
```

**Comment by atjbramley** [comment]
Hi, 

I'm trying to produce visualisations just like the timeseries graph in your first image.

When I set the connection options and Parsing options just as you have them in the screenshot, I am unable to produce a time-series graph as I get a 'Data is missing a number field' error.

The data shows up just fine in Table view.

Please could you let me know where I'm going wrong?

**Reply by yesoreyeram** [reply]
AWS status feed is not a timeseries data. so can't directly visualise it in timeseries graph. The timeseries screenshot I used is a representation of AWS annotation.

**Reply by atjbramley** [reply]
Thanks - I think I realised this some time after posting the issue and should have closed this.

**Comment by eugenesheyko** [comment]
I'm trying to parse the [status.aws.amazon.com/rss/all.rss](https://status.aws.amazon.com/rss/all.rss) RSS feed as suggested. It doesn't look like it work if we specify the "rss.channel[0].item" value as Rows/Root but it works if set the "rss.channel", this way looks like it can be set only for the specific service as a single row.
![health_status_dashboards_grafana](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/19654539/3527b08f-32bd-4bff-a49c-4fbc87d9c040)

**Comment by spongenee** [comment]
Ditto that following the tutorial does not yield the feed table anymore. this is the current [status.aws.amazon.com/rss/all.rss](https://status.aws.amazon.com/rss/all.rss)

<img width="987" alt="image" src="https://github.com/user-attachments/assets/df4cf3a5-dc04-4b74-877c-0027e673f708">
